rocAL/source/readers/file_source_reader.cpp

```diff
@@ -78,11 +71,18 @@ Reader::Status FileSourceReader::initialize(ReaderConfig desc) {
 void FileSourceReader::incremenet_read_ptr() {
     _read_counter++;
     _curr_file_idx = (_curr_file_idx + 1) % _file_names.size();
+    if (_last_batch_info.first == RocalBatchPolicy::DROP) {
+        if ((_file_names.size() / _batch_count) == _curr_file_idx)  // Check if its last batch
+        {
+            _curr_file_idx += _batch_count;
+            _curr_file_idx = (_curr_file_idx + 1) % _file_names.size();
```

Comment:
_curr_file_idx += _batch_count; // This increaments the ptr with batch size - meaning the batch is skipped. 
 _curr_file_idx = (_curr_file_idx + 1) % _file_names.size();
 // When the last_batch_pad is true, next iter should start from beginning.
 This line ensures, pointer from end -> is brought back to begging.
 
We have not followed your changes where we do not pad the file_names. Curently the logic is done by padding the file_names vector , hence should work for True.

We will add a note stating, this will not work "last_batch_pad=False" case.

Reply:
Please add the note

Reply:
Also add comments for both the lines stating what it does,

_curr_file_idx += _batch_count;
            _curr_file_idx = (_curr_file_idx + 1) % _file_names.size();